### PR TITLE
[refactor] observer patterns

### LINKED ISSUE #120: refactor ItemHud
- implement boolean on ItemHud
- make UpdateDisplay function

### LINKED ISSUE #139: Refactor ItemHUD and overworld item (change to observer)
https://onewheelstudio.com/blog/2020/7/24/observer-pattern-c-events

### PR DESCRIPTION
**Pull request by raihansyahdean**
### Main changes
- Rafactor audio pattern
- Overworld Item class
- Volume class
- Audio manager class (not observer, but find child so no longer using manual assignment)

### Tasks List
- [close #120 ]
- [close #139 ]

### Attachments
image
image

### New Packages
N/A

### Resources 
N/A

### CODE REVIEW

**Review by VadimBhirawa — APPROVED** [review]
Looks good!

**Review by naufalhilmi12345 — APPROVED** [review]
Cool! LGTM

**Review by RaiRiz — APPROVED** [review]
LGTM!

File: Assets/Astralization/Scripts/Interactables/Interactable.cs [hunk 1]
```diff
@@ -26,13 +27,17 @@ public abstract class Interactable : MonoBehaviour, IInteractable
     [Space]
     [Header("Audio")]
     [Tooltip("Audio Manager")]
-    public AudioManager AudioManager;
+    protected AudioManager _audioManager;
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Interactables/Interactable.cs:30]: protected field use PascalCase

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 2]
```diff
@@ -160,15 +160,19 @@ public void DiscardItem(InputAction.CallbackContext ctx)
...
     {
-        float scrollValue = ctx.ReadValue<float>();
-        int indexShift = (int)scrollValue / (_scrollStep * ScrollSensitivity);
-        int newIdx = Utils.MathCalcu.mod(_activeIdx - indexShift, InvenLength);
-        ChangeActiveItem(newIdx);
+        if (ctx.performed)
+        {
+            Vector2 scrollVector = ctx.ReadValue<Vector2>();
+            float scrollValue = scrollVector.y;
+            int indexShift = (int) (scrollValue / (_scrollStep * ScrollSensitivity));
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Inventory.cs:170]: use GetScrollStep

**Review by rafiftaris — CHANGES_REQUESTED** [review]

File: Assets/Astralization/Scripts/Player/Inventory.cs [hunk 3]
```diff
@@ -167,7 +167,7 @@ public void ScrollActiveItem(InputAction.CallbackContext ctx)
         {
             Vector2 scrollVector = ctx.ReadValue<Vector2>();
             float scrollValue = scrollVector.y;
-            int indexShift = (int) (scrollValue / (_scrollStep * ScrollSensitivity));
+            int indexShift = (int) (scrollValue / (GetScrollStep()));
```
Comment (rafiftaris) [inline Assets/Astralization/Scripts/Player/Inventory.cs:170]: GetScrollStep gausah dikurungin

**Review by rafiftaris — CHANGES_REQUESTED** [review]

**Review by rafiftaris — APPROVED** [review]
mantap dah

**Pull request merged**